infra/storage/kFileTransferMgr.class.php

```diff
@@ -445,21 +445,35 @@ public function getFile ( $remote_file, $local_file = null)
 		$this->results = $res;
 		
 		// check response
-		if ( ! $res ) {
+		if ( ! $res )
+		{
 			$last_error = error_get_last();
 			throw new kFileTransferMgrException("Can't get file [$remote_file] - " . $last_error['message'], kFileTransferMgrException::otherError);
-			return self::FILETRANSFERMGR_RES_ERR;
 		}
 		else
 		{
 			KalturaLog::debug("File retrieved successfully");
 			if(is_null($local_file))
 				return $res;
+
+			$this->validateFileSize(strlen($res), $local_file);
 				
 			return self::FILETRANSFERMGR_RES_OK;
 		}
 	}
 
+	private function validateFileSize( $expectedSize ,$localFile)
+	{
+		clearstatcache();
+		$actualFileSize = kFile::fileSize($localFile);
+		if($expectedSize && $actualFileSize < $expectedSize)
```

Comment:
Lets check $expectedSize in outer scope , since if the value is 0 , no since in checking actual file size and clearcache.